Plugins/NativeCapture/Source/NativeCapture.cpp

```diff
@@ -34,19 +142,34 @@ namespace Babylon::Plugins::Internal
         NativeCapture(const Napi::CallbackInfo& info)
             : Napi::ObjectWrap<NativeCapture>{info}
             , m_runtime{JsRuntime::GetFromJavaScript(info.Env())}
-            , m_graphicsImpl(Graphics::Impl::GetFromJavaScript(info.Env()))
-            , m_ticket{std::make_unique<TicketT>(m_graphicsImpl.AddCaptureCallback([this](auto& data) { CaptureDataReceived(data); }))}
             , m_jsData{Napi::Persistent(Napi::Object::New(info.Env()))}
         {
+            auto& graphicsImpl{Graphics::Impl::GetFromJavaScript(info.Env())};
+
             Napi::Object jsData = m_jsData.Value();
             jsData.Set("data", Napi::ArrayBuffer::New(info.Env(), 0));
+
+            FrameCallback frameCallback{[this](uint32_t width, uint32_t height, bgfx::TextureFormat::Enum format, bool yFlip, gsl::span<const uint8_t> data){
+                CaptureDataReceived(width, height, format, yFlip, data);
+            }};
+
+            bgfx::FrameBufferHandle frameBufferHandle{bgfx::kInvalidHandle};
+
+            // For an off screen frame buffer, the frame buffer must be passed into the constructor. Otherwise, we capture the on screen / default frame buffer.
+            if (info.Length() > 0 && info[0].IsExternal())
```

Comment:
The IsExternal should probably throw. Otherwise, we are just letting invalid arguments go through.